### PR TITLE
Use a tmpfs for /var

### LINKED ISSUE #2288: fully immutable filesystem
Currently a few parts of the init script cheat and make filesystem modifiable. Fix these

- [x] dhcp /etc/resolv/conf see #1920 
- [ ] /var if not overmounted, should be a tmpfs
- [x] container root filesystem, should add a tmpfs overlay if writeable #2309

At this point we should be able to mount the filesystem `ro` very early, and run off a cdrom filesystem or similar too if needed.

### PR DESCRIPTION
**Pull request by justincormack**
Previously we were cheating and remounting /var `rw` but this does not
work if the filesystem is really read only. Nount a tmpfs, which may
be overmounted later by a persistent filesystem.

Signed-off-by: Justin Cormack <justin.cormack@docker.com>

fix #2288 

racoon

### CODE REVIEW

**Comment by justincormack** [comment]
Will upload new hashes shortly

**Comment by justincormack** [comment]
hash updates added

**Comment by justincormack** [comment]
Hmm, bunch of CI failures, including the `extend` ones and others:

```
[+ 49m 57s] [FAIL    ] linuxkit.packages.getty-containerd 105.39s
[+ 51m 00s] [FAIL    ] linuxkit.packages.format_mount.auto 63.06s
[+ 57m 20s] [FAIL    ] linuxkit.packages.format_mount.by_label 380.72s
[+ 58m 46s] [PASS    ] linuxkit.packages.format_mount.by_name 85.64s
[+ 60m 16s] [FAIL    ] linuxkit.packages.format_mount.btrfs 90.20s
[+ 61m 13s] [FAIL    ] linuxkit.packages.format_mount.xfs 57.11s
[+ 62m 32s] [FAIL    ] linuxkit.packages.format_mount.multiple 78.53s
[+ 70m 14s] [FAIL    ] linuxkit.packages.extend.ext4 461.74s
[+ 74m 46s] [FAIL    ] linuxkit.packages.extend.btrfs 272.79s
[+ 80m 33s] [FAIL    ] linuxkit.packages.extend.xfs 346.43s
[+ 87m 03s] [PASS    ] linuxkit.packages.mkimage 389.84s
[+ 89m 27s] [PASS    ] linuxkit.packages.sysctl 144.62s
```

**Comment by justincormack** [comment]
Hmm, well the `containerd` test is failing with `containerd: mkdir /var/lib/containerd: read-only file system`

**Comment by justincormack** [comment]
Some issues in tests that are conflicting with changes in #2309, sigh, will wait for that to pass first.

**Comment by justincormack** [comment]
Ok, found a bug in directories permissions in `mountie`, which account for most of the test failures.

**Comment by justincormack** [comment]
ok, one failure left to resolve...

**Comment by justincormack** [comment]
Ok, this is fixed now.

File: pkg/init/bin/rc.init [hunk 1]
```diff
@@ -1,11 +1,29 @@
...
+mount -n -t tmpfs tmpfs /var -o nodev,nosuid,noexec,relatime,size=50%,mode=755
+# add standard directories in /var
+mkdir -m 755 /var/cache
+mkdir -m 555 /var/empty
+mkdir -m 755 /var/lib
+mkdir -m 755 /var/local
+mkdir -m 755 /var/lock
+mkdir -m 755 /var/log
+mkdir -m 755 /var/opt
+ln -s /run /var/run
```
Comment (riyazdf) [inline pkg/init/bin/rc.init]: 👍  nice

**Pull request merged**